test/deploy/kubernetes/deploy.sh

```diff
@@ -83,6 +83,12 @@ kubectl apply -f "$BASEPATH/external-services/headless_control_node.yaml" --name
 print_header "Creating Cotrol Node - Cluster"
 kubectl apply -f "$BASEPATH/provisioner-pods/deployment_control_node.yaml" --namespace "$NAMESPACE"
 
+# Wait for Control-POD execution 
+while [ "$(kubectl get pods | grep control-node | awk '{print $3}')" != "completed" ]; do
```

Comment:
This will keep running forever if status of control-node pod fails, right?
If so, you can add logic to fail the deploy script after maximum no of attempts, say 100

`        try=1; max_tries=100
        until  kubectl get pods | grep control-node | awk '{print $3}') != "completed" 
        do
            if [[ "$try" -gt "$max_tries" ]]; then
                echo "Error, exiting" 
                exit 1
            fi
            try=$(( $try + 1 ))
            # Wait for 20 sec
            sleep 20
        done
`